src/Raven.Server/Documents/Indexes/Persistence/Corax/AnonymousCoraxDocumentConverter.cs

```diff
@@ -24,88 +22,68 @@ public abstract class AnonymousCoraxDocumentConverterBase : CoraxDocumentConvert
 
                 
-                InsertRegularField(field, value, indexContext, ref entryWriter, sourceDocument, scope, out var innerShouldSkip);
-                shouldSkip &= innerShouldSkip;
+            InsertRegularField(field, value, indexContext, builder, sourceDocument, out var innerShouldSkip);
+            hasFields = true;
                 
-                
-                if (storedValue is not null && innerShouldSkip == false)
-                {
-                    //Notice: we are always saving values inside Corax index. This method is explicitly for MapReduce because we have to have JSON as the last item.
-                    var blittableValue = TypeConverter.ToBlittableSupportedType(value, out TypeConverter.BlittableSupportedReturnType returnType, flattenArrays: true);
-
-                    if (returnType != TypeConverter.BlittableSupportedReturnType.Ignored)
-                        storedValue[property.Key] = blittableValue;
-                }
-            }
-
-            if (shouldSkip && entryWriter.IsEmpty() && _indexEmptyEntries == false)
-            {
-                fields = 0;
-                output = default;
-                return default;
-            }
-            
-            if (storedValue is not null)
+            if (storedValue is not null && innerShouldSkip == false)
             {
-                var bjo = indexContext.ReadObject(storedValue, "corax field as json");
-                scope.Write(string.Empty, knownFields.Count - 1, bjo, ref entryWriter);
+                //Notice: we are always saving values inside Corax index. This method is explicitly for MapReduce because we have to have JSON as the last item.
+                var blittableValue = TypeConverter.ToBlittableSupportedType(value, out TypeConverter.BlittableSupportedReturnType returnType, flattenArrays: true);
+
+                if (returnType != TypeConverter.BlittableSupportedReturnType.Ignored)
+                    storedValue[property.Key] = blittableValue;
             }
-            
-            id = key ?? throw new InvalidParameterException("Cannot find any identifier of the document.");
-            if (sourceDocumentId != null && knownFields.TryGetByFieldName(Constants.Documents.Indexing.Fields.SourceDocumentIdFieldName, out var documentSourceField))
-                scope.Write(string.Empty, documentSourceField.FieldId, sourceDocumentId.AsSpan(), ref entryWriter);
-            
-            scope.Write(string.Empty, 0, id.AsSpan(), ref entryWriter);
-            fields = entryWriter.CurrentFieldCount();
-            return entryWriter.Finish(out output, _indexImplicitNull);
         }
-        catch
+
+        if (storedValue is not null)
         {
-            ResetEntriesWriter();
-            throw;
+            var bjo = indexContext.ReadObject(storedValue, "corax field as json");
```

Comment:
is there a place where we can dispose this?

Reply:
Now there is :-)